pex/environment.py

```diff
@@ -219,11 +219,10 @@ def can_add(self, dist):
       # platforms it supports at buildtime and runtime so this is always safe.
       return True
 
-    try:
-      wheel_name_, wheel_raw_version_, wheel_tags = filename.split('-', 2)
-    except ValueError:
-      return False
-
+    # Wheel filename format: https://www.python.org/dev/peps/pep-0427/#file-name-convention
+    # `{distribution}-{version}(-{build tag})?-{python tag}-{abi tag}-{platform tag}.whl`
+    wheel_parts = filename[:-4].split('-')
```

Comment:
This is the red in CI - filename from above already has `'.whl'` stripped.

Reply:
🤦 that's why! i was testing in console and missed that

Reply:
I helped out with the last road bump here.